backend/open_api_spec.yml

```diff
@@ -0,0 +1,790 @@
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              additionalProperties: false
+              properties:
+                curator_tag:
+                  type: string
+                  description: curator-provided tag
+      responses:
+        "204":
+          $ref: "#/components/responses/204"
+        "400":
+          $ref: "#/components/responses/400"
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+
+  /curation/collections/{collection_id}/datasets/upload-links:
+    put:
+      description: Upload a Dataset. The Dataset will be downloaded from the provided link.
+      security:
+        - cxguserCookie: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              additionalProperties: false
+              properties:
+                curator_tag:
+                  type: string
+                id:
+                  $ref: "#/components/schemas/dataset_uuid"
+                link:
+                  $ref: "#/components/schemas/upload_link"
+      responses:
+        "202":
+          $ref: "#/components/responses/202"
+        "400":
+          $ref: "#/components/responses/400"
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+
+  /curation/collections/{collection_id}/datasets/upload-s3:
+    put:
+      description: >
+        Retrieve AWS credentials for uploading Dataset source files to S3 to create
+        or update Datasets for the specified Collection.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  credentials:
+                    type: object
+                    properties:
+                      access_key:
+                        type: string
+                        required: true
+                      secret_access_key:
+                        type: string
+                        required: true
+                      access_token:
+                        type: string
+                        required: true
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+
+  /curation/collections/{collection_id}/datasets/status:
+    get:
+      description: Get the processing status for a Dataset.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+        - $ref: "#/components/parameters/optional_query_curator_tag"
+        - $ref: "#components/parameters/optional_query_dataset_uuid"
```

Comment:
If the dataset UUID is an optional parameter, the response should probably be an array with one status per dataset.